### ISSUE TITLE
Many deprecated notes when using PivotX 3.0 rc3 under PHP 8.2.7

### ISSUE THREAD

**Issue by Lemuria3** [pivotx/PivotX#16]
I have downloaded the latest PivotX 3.0 rc3 dated 23 June 2023 and uploaded it to the test web server.

Environment: 

Web server: Apache/2.4 Server API CGI/FastCGI
Server default charset:	UTF-8
PHP Version: 8.2.7
Error Reporting Value: 32767

When calling up the URL https://domain.xyz/pivotx, the following errors occur:

Deprecated: Optional parameter $type declared before required parameter $error_msg is implicitly treated as a required parameter in ../pivotx/pivotx/lib.php on line 458
Deprecated: Using ${var} in strings is deprecated, use {$var} instead in ../pivotx/pivotx/lib.php on line 614
Deprecated: Using ${var} in strings is deprecated, use {$var} instead in ../pivotx/pivotx/lib.php on line 628
Deprecated: Using ${var} in strings is deprecated, use {$var} instead in ../pivotx/pivotx/lib.php on line 2530
Deprecated: Using ${var} in strings is deprecated, use {$var} instead in ../pivotx/pivotx/lib.php on line 2542
Deprecated: Optional parameter $uid declared before required parameter $number is implicitly treated as a required parameter in ../pivotx/pivotx/lib.php on line 4037
Deprecated: Optional parameter $uid declared before required parameter $number is implicitly treated as a required parameter in ../pivotx/pivotx/lib.php on line 4064
Deprecated: Optional parameter $uid declared before required parameter $number is implicitly treated as a required parameter in ../pivotx/pivotx/lib.php on line 5506
Deprecated: Optional parameter $force declared before required parameter $unit is implicitly treated as a required parameter in ../pivotx/pivotx/modules/module_db.php on line 100
Deprecated: Optional parameter $max declared before required parameter $match is implicitly treated as a required parameter in ../pivotx/pivotx/modules/module_tags.php on line 95
Deprecated: Optional parameter $amount declared before required parameter $comments is implicitly treated as a required parameter in ../pivotx/pivotx/modules/module_parser.php on line 1787
Deprecated: Optional parameter $only declared before required parameter $searchresults is implicitly treated as a required parameter in ../pivotx/pivotx/modules/module_search.php on line 434
Deprecated: Optional parameter $weblog declared before required parameter $searchresults is implicitly treated as a required parameter in ../pivotx/pivotx/modules/module_search.php on line 434
Deprecated: Optional parameter $category declared before required parameter $searchresults is implicitly treated as a required parameter in ../pivotx/pivotx/modules/module_search.php on line 434
Deprecated: Optional parameter $weblogname declared before required parameter $subweblogname is implicitly treated as a required parameter in ../pivotx/pivotx/objects.php on line 1161
Deprecated: Optional parameter $type declared before required parameter $value is implicitly treated as a required parameter in ../pivotx/pivotx/objects.php on line 3166
Deprecated: Optional parameter $type declared before required parameter $values is implicitly treated as a required parameter in ../pivotx/pivotx/objects.php on line 3197
Deprecated: Optional parameter $type declared before required parameter $key is implicitly treated as a required parameter in ../pivotx/pivotx/objects.php on line 3220
Deprecated: Using ${var} in strings is deprecated, use {$var} instead in ../pivotx/pivotx/modules/module_smarty.php on line 2910
Deprecated: Optional parameter $pDirectory declared before required parameter $pFieldName is implicitly treated as a required parameter in ../pivotx/pivotx/modules/formclass.php on line 1232
Deprecated: Optional parameter $pMaxFileSize declared before required parameter $pFieldName is implicitly treated as a required parameter in ../pivotx/pivotx/modules/formclass.php on line 1232
Deprecated: Optional parameter $error_msg declared before required parameter $error_no is implicitly treated as a required parameter in ../pivotx/pivotx/modules/module_sql.php on line 199
Deprecated: Function utf8_decode() is deprecated in ../pivotx/pivotx/lib.php on line 4527
Deprecated: Function utf8_decode() is deprecated in ../pivotx/pivotx/lib.php on line 4527

The PivotX blog is not displayed in the browser window, only the errors are visible.

**Comment by hansfn** [comment]
Thx for the report, but I haven't time to fix all deprecations. Feel free to submit a PR ;-) I might fix some eventually.

Please change `error_reporting` from `E_ALL` to `E_ALL & ~E_DEPRECATED & ~E_STRICT` so we can see if there is some "real" error causing the blog to not display. You have checked the webserver error log?

**Comment by Lemuria3** [comment]
Ok, I have changed the Test Web Server to the Production Value error_reporting, deleted the existing PivotX installation and performed a completely new upload.
The setup appears, I can create a user and the PivotX blog appears in the browser window without PHP errors.  Then I quickly wrote a blog entry and it seems to work. Again, no PHP error.

After that I set the Test Web Server back to Development Value, cleared the browser cache completely, cleared the PivotX cache as well and the PivotX blog appears in the browser window this time, but with many deprecated errors (see above).

For your information,
the Apache server log does not show any errors that refer to the PivotX installation.
